Src/BootCamp.Chapter/CleanTable.cs

```diff
@@ -0,0 +1,171 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Text;
+
+namespace BootCamp.Chapter
+{
+    class CleanTable
+    {
+        private const char DollarSign = '£';
+        private string _corruptedBalanceFile;
+        private string _fixedBalanceFile;
+
+        public CleanTable(string corruptedBalanceFile, string fixedBalanceFile)
+        {
+            if (!File.Exists(corruptedBalanceFile))
+            {
+                throw new ArgumentException($"The corrupted balance file does not exist: {corruptedBalanceFile}.");
+            }
+            _corruptedBalanceFile = corruptedBalanceFile;
+
+            if (File.Exists(fixedBalanceFile))
+            {
+                File.Delete(fixedBalanceFile);
+            }
+            _fixedBalanceFile = fixedBalanceFile;
+        }
+
+        public void Clean()
+        {
+            var corruptedBalanceLines = File.ReadAllLines(_corruptedBalanceFile);
+            var fixedBalanceLines = new string[corruptedBalanceLines.Length];
+            FixCorruptedBalanceFile(corruptedBalanceLines, fixedBalanceLines);
+
+            DisplayStasticalData(CreateBalanceLinesForTextTable(fixedBalanceLines));
+        }
+
+        private string[] CreateBalanceLinesForTextTable(string[] fixedBalanceLines)
+        {
+            var balanceLinesForTextTable = new string[fixedBalanceLines.Length];
+            var balanceSeperator = ",";
+
+            for (var i = 0; i < fixedBalanceLines.Length; i++)
+            {
+                var balances = fixedBalanceLines[i].Split(balanceSeperator);
+                var balanceLine = new StringBuilder(balances[0]);
+
+                for (var j = 1; j < balances.Length; j++)
+                {
+                    balanceLine.Append(", ");
+                    balanceLine.Append(balances[j].Replace(DollarSign.ToString(), ""));
+                }
+
+                balanceLinesForTextTable[i] = balanceLine.ToString();
+            }
+
+            return balanceLinesForTextTable;
+        }
+
+        private void FixCorruptedBalanceFile(string[] corruptedBalanceLines, string[] fixedBalanceLines)
+        {
+            var buildLines = new StringBuilder();
+            var skipCarriageReturn = corruptedBalanceLines.Length - 1;
+
+            for (var i = 0; i < corruptedBalanceLines.Length; i++)
+            {
+                fixedBalanceLines[i] = corruptedBalanceLines[i].Replace("_", "");
+
+                if (!isValidLine(fixedBalanceLines[i]))
+                {
+                    throw new Exception("Invalid Balance.");
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: ['System.Exception' should not be thrown by user code.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=5063788)